smart_open/smart_open_lib.py

```diff
@@ -551,8 +562,8 @@ def __init__(self, outkey, min_part_size=S3_MIN_PART_SIZE, **kw):
         if not hasattr(outkey, "bucket") and not hasattr(outkey, "name"):
             raise TypeError("can only process S3 keys")
 
-        if is_gzip(outkey.name):
-            raise NotImplementedError("streaming write to S3 gzip not supported")
+        # if is_gzip(outkey.name):
```

Comment:
Removing the check is a temporary solution while waiting for proper fix described by @mpenkov in #82 for all combinations of `mode="r/w", native_gzip=True/False`